### PR TITLE
feat: Travis CI for building

### PR DESCRIPTION
**Pull request by danjohnson95**
This PR cleans up the way the documentation is built.

I've created a Docker container which builds and serves the files, so you don't need to install Hugo.

## Serving Locally

```bash
$ docker-compose up
```

will start up the webserver on http://localhost:1313.

## Building Manually

Building happens automatically on deployment by Travis, so you **don't** need to run this. But it might come in handy if you wanted the built files 🤷‍♂️ 

```bash
$ make
```

will build the files and output them to `dist/`.

`dist/` is .gitignore'd, so you won't be able to manually deploy by mistake.

### What is `make` doing?

**`docker-compose run hugo hugo`**
Runs the `hugo` container, and specifies that the `hugo` command should be run (instead of `hugo server`). This builds the docs and outputs them to `/dist`. 

**`docker-compose run assets npm run build`**
Runs the `assets` container which will install javascript dependencies and run gulp build task. Assets will be outputted inside the `/dist` directory.

## Deployment

When changes are pushed to `master`, Travis will run the `deploy` part of the .travis.yml config file. This will:
  * Clone the repository and checkout the master branch
  * Run `make`
  * Remove the `.gitignore` from `/dist` (so that we can commit the contents)
  * Creates a `CNAME` file using the FQDN specified in the Travis config
  * Commits the contents of `/dist` into the `gh-pages` branch*

\* Using the `gh-pages` branch means that Travis will never commit to master, so we don't have to worry about conflicts or overwritten data.

## TL;DR
* You no longer need to commit the build files
* Pushing to master will deploy straight away
* You need to install Docker
* Run `docker-compose up` to serve docs locally

## TODO
* Switch over the GitHub pages destination from `master` to `gh-pages`.
* Make `master` a protected branch

### CODE REVIEW

**Comment by StevePorter92** [comment]
This is awesome

**Comment by danjohnson95** [comment]
fyi @designmynight/all-developers
